src/scripts/OutSystems/OSUI/Utils/Menu.ts

```diff
@@ -123,31 +125,34 @@ namespace OutSystems.OSUI.Utils.Menu {
 
 	// OnResize handler
 	const _onResizeCallbackHandler = (): void => {
-		// Get the current device type
-		const currentDeviceType = OSFramework.OSUI.Helper.DeviceInfo.GetDeviceType();
-
-		// Check if the device type is the same as the current device type, if so, return to prevent unnecessary calls
-		if (_appProp.device.type === currentDeviceType) {
-			return;
-		}
+		window.clearTimeout(_onResizeTimeout);
+		_onResizeTimeout = OSFramework.OSUI.Helper.ApplySetTimeOut(() => {
+			// Get the current device type
+			const currentDeviceType = OSFramework.OSUI.Helper.DeviceInfo.GetDeviceType();
+
+			// Check if the device type is the same as the current device type, if so, return to prevent unnecessary calls
+			if (_appProp.device.type === currentDeviceType) {
+				return;
+			}
 
-		// Update the device type on the app properties
-		_appProp.device.type = currentDeviceType;
+			// Update the device type on the app properties
+			_appProp.device.type = currentDeviceType;
 
-		// Check if the menu is open
-		if (_appProp.menu.isOpen) {
-			// Close the menu, internally it will update the menu attributes
-			_toggleMenu();
-		} else {
-			// Update app properties and menu attributes
-			_updatePropsAndAttrs();
-		}
+			// Check if the menu is open
+			if (_appProp.menu.isOpen) {
+				// Close the menu, internally it will update the menu attributes
+				_toggleMenu();
+			} else {
+				// Update app properties and menu attributes
+				_updatePropsAndAttrs();
+			}
 
-		// Remove the menu event listeners since device type changed
-		_removeMenuEventListeners();
+			// Remove the menu event listeners since device type changed
+			_removeMenuEventListeners();
 
-		// ReAdd the menu event listeners
-		_addMenuEventListeners();
+			// ReAdd the menu event listeners
+			_addMenuEventListeners();
+		}, 300);
```

Comment:
@JoaoFerreira-FrontEnd, I would say we should left a comment here about why these 300ms are needed here.

Reply:
added;